File: esports-backend/src/main/java/com/esports/dto/ProjectCreateDTO.java
package com.esports.dto;

import jakarta.validation.constraints.Future;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
public class ProjectCreateDTO {
    @NotBlank(message = "Title is required")
    @Pattern(regexp = "^[a-zA-Z\\s]+$", message = "Title must contain only alphabets and spaces")
    @Size(max = 20, message = "Title must not exceed 20 characters")
    private String title;

    @NotBlank(message = "Description is required")
    @Size(max = 300, message = "Description must not exceed 300 characters")
    private String description;

    @NotNull(message = "Service ID is required")
    private Long serviceId;

    @NotNull(message = "Deadline is required")
    @Future(message = "Deadline must be in the future")
    private LocalDateTime deadline;

    @NotNull(message = "At least 2 milestones are required")
    @Size(min = 2, message = "At least 2 milestones are required")
    private List<String> milestones;
}

File: esports-backend-dotnet/Upkart1/DTO/ProjectCreateDTO.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;
using Upkart1.Validation;

namespace Upkart1.DTO
{
    public class ProjectCreateDTO
    {
        [Required(ErrorMessage = "Title is required")]
        [RegularExpression(@"^[a-zA-Z\s]+$", ErrorMessage = "Title must contain only alphabets and spaces")]
        [StringLength(20, ErrorMessage = "Title must not exceed 20 characters")]
        [JsonPropertyName("title")]
        public string Title { get; set; }

        [Required(ErrorMessage = "Description is required")]
        [StringLength(300, ErrorMessage = "Description must not exceed 300 characters")]
        [JsonPropertyName("description")]
        public string Description { get; set; }

        [Required(ErrorMessage = "Service ID is required")]
        [JsonPropertyName("serviceId")]
        public long ServiceId { get; set; }

        [Required(ErrorMessage = "Deadline is required")]
        [FutureDateString(ErrorMessage = "Deadline must be in the future")]
        [JsonPropertyName("deadline")]
        public string Deadline { get; set; }

        [Required(ErrorMessage = "At least 2 milestones are required")]
        [MinLength(2, ErrorMessage = "At least 2 milestones are required")]
        [JsonPropertyName("milestones")]
        public List<string> Milestones { get; set; }
    }
}